src/mon/OSDMonitor.cc

```diff
@@ -6417,8 +6512,14 @@ int OSDMonitor::prepare_new_pool(string& name, uint64_t auid,
   pi->crush_rule = crush_rule;
   pi->expected_num_objects = expected_num_objects;
   pi->object_hash = CEPH_STR_HASH_RJENKINS;
-  pi->set_pg_num(pg_num);
-  pi->set_pgp_num(pgp_num);
+  auto max = g_conf->get_val<int64_t>("mon_osd_max_initial_pgs");
+  pi->set_pg_num(
+    max > 0 ? std::min<uint64_t>(pg_num, std::max<int64_t>(1, max))
+    : pg_num);
```

Comment:
Maybe.. i don't think this is so different than the fact that all pgs aren't created at once.  The difference is that instead of the mon creating 100 at a time, the mon creates the initial (max) and then the osds split them out from there. I don't think there is any functional change from the users perspective.

And I'd also like to get away from the habit of having side-band communication to the (human) user via the stderr stream in the response.